pootle/apps/pootle_store/abstracts.py

```diff
@@ -31,6 +32,36 @@
 fs = PootleFileSystemStorage()
 
 
+class AbstractUnitSource(models.Model):
+
+    class Meta(object):
+        abstract = True
+
+    unit = models.ForeignKey(
+        "pootle_store.Unit",
+        db_index=True,
+        unique=True,
+        null=False,
+        blank=False,
+        related_name="unit_source",
+        on_delete=models.CASCADE)
+
+    created_by = models.ForeignKey(
+        settings.AUTH_USER_MODEL,
+        null=False,
+        blank=False,
+        db_index=True,
+        related_name='created_units',
+        default=get_system_user_id,
+        on_delete=models.SET(get_system_user))
+
+    created_with = models.IntegerField(
```

Comment:
it will be used to record how the unit was created - ie `UPLOAD` `FS` `CLONE` etc